src/spikeinterface/postprocessing/template_similarity.py

```diff
@@ -63,6 +64,63 @@ def _select_extension_data(self, unit_ids):
         new_similarity = self.data["similarity"][unit_indices][:, unit_indices]
         return dict(similarity=new_similarity)
 
+    def _merge_extension_data(
+        self, units_to_merge, new_unit_ids, new_sorting_analyzer, keep_mask=None, verbose=False, **job_kwargs
+    ):
+        num_shifts = int(self.params["max_lag_ms"] * self.sorting_analyzer.sampling_frequency / 1000)
+        templates_array = get_dense_templates_array(
+            new_sorting_analyzer, return_scaled=self.sorting_analyzer.return_scaled
+        )
+        arr = self.data["similarity"]
+        sparsity = new_sorting_analyzer.sparsity
+        all_new_unit_ids = new_sorting_analyzer.unit_ids
+        new_similarity = np.zeros((len(all_new_unit_ids), len(all_new_unit_ids)), dtype=arr.dtype)
+
+        for unit_id1 in all_new_unit_ids:
+            unit_ind1 = new_sorting_analyzer.sorting.id_to_index(unit_id1)
+            template1 = templates_array[unit_ind1][np.newaxis, :]
+            if sparsity is not None:
+                sparsity1 = ChannelSparsity(sparsity.mask[unit_ind1][np.newaxis, :], [unit_id1], sparsity.channel_ids)
+            else:
+                sparsity1 = None
+            if unit_id1 in new_unit_ids:
+                new_spk1 = True
+            else:
+                new_spk1 = False
+                i = self.sorting_analyzer.sorting.id_to_index(unit_id1)
+
+            for unit_id2 in all_new_unit_ids[unit_ind1:]:
+                unit_ind2 = new_sorting_analyzer.sorting.id_to_index(unit_id2)
+                template2 = templates_array[unit_ind2][np.newaxis, :]
+                if sparsity is not None:
+                    sparsity2 = ChannelSparsity(
+                        sparsity.mask[unit_ind2][np.newaxis, :], [unit_id2], sparsity.channel_ids
+                    )
+                else:
+                    sparsity2 = None
+                if unit_id2 in new_unit_ids:
+                    new_spk2 = True
+                else:
+                    new_spk2 = False
+                    j = self.sorting_analyzer.sorting.id_to_index(unit_id2)
+
+                if new_spk1 or new_spk2:
+                    new_similarity[unit_ind1, unit_ind2] = compute_similarity_with_templates_array(
+                        template1,
+                        template2,
+                        method=self.params["method"],
+                        num_shifts=num_shifts,
+                        support=self.params["support"],
+                        sparsity=sparsity1,
+                        other_sparsity=sparsity2,
+                    )
+                else:
+                    new_similarity[unit_ind1, unit_ind2] = arr[i, j]
```

Comment:
we should have the same machinery for run and merge oerthwise bug and improvement will not be sync.

Reply:
This is the same function called internally. I'm ok to recompute everything (as done with CCGs), but this is slower when you have only few merges